src/emulator/storage/files.ts

```diff
@@ -166,8 +301,9 @@ export class StorageLayer {
         request.authorization
       );
     }
+
     if (!authorized) {
-      throw new ForbiddenError("Failed auth");
+      throw new ForbiddenError("Permission denied. No READ permission");
```

Comment:
For the permission denied errors, we can omit the error message. This is done elsewhere.